eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/ReplaceableTxPublisher.scala

```diff
@@ -329,6 +329,14 @@ private class ReplaceableTxPublisher(nodeParams: NodeParams,
     Behaviors.receiveMessagePartial {
       case WrappedTxResult(MempoolTxMonitor.TxConfirmed) => sendResult(replyTo, TxPublisher.TxConfirmed(cmd, tx))
       case WrappedTxResult(MempoolTxMonitor.TxRejected(reason)) =>
+        reason match {
+          case TxRejectedReason.WalletInputGone =>
+            // The transaction now has an unknown input from bitcoind's point of view, so it will keep it in the wallet in
+            // case that input appears later in the mempool or the blockchain. In our case, we know it won't happen so we
+            // abandon that transaction and will retry with a different set of inputs (if it still makes sense to publish).
+            bitcoinClient.abandonTransaction(tx.txid)
+          case _ => // nothing to do
+        }
```

Comment:
Shouldn't there be a check here that the remote commit has been published, instead of only relying on the error code from bitcoind?

Reply:
I don't think so, it's more general than that.
This case can also happen for htlc txs that spend an unconfirmed output of another htlc tx which is then replaced.
I think that in all cases where one of our wallet input disappears like this, we should abandon the transaction to free the inputs, and we'll retry later if needed anyway so there's no risk that we forget to claim an output.